### ISSUE TITLE
Preserve existing terminal input when sending code to termina

### ISSUE THREAD

**Issue by assignUser** [Ikuyadeu/vscode-R#400]
Situation: You have started to type something into the terminal (here `library(lintr)`) but switch back to the editor before clearing the line/sending the command. You now send some code from the editor with `ctrl+enter` (here `x <- 1:10`).

In Rstudion the unfinished input is preserved and the code from the editor is run without issue:
![image](https://user-images.githubusercontent.com/16141871/90524451-5d86f800-e16e-11ea-9f32-1d341b4c4f92.png)

In vscode-R the code from the editor is pasted into the terminal without regard for any prior input, which will ofc cause the commands to fail (or unexpected behavior if valid):
![image](https://user-images.githubusercontent.com/16141871/90524589-8b6c3c80-e16e-11ea-9468-f801d6b6aabb.png)

I am not sure if this would be possible in vsc or if extensions cant modify terminals in that way, but emulating the behavior of rstudio would make working with vsc-r even smoother ❤️

**Comment by renkun-ken** [comment]
This is a bit annoying for me too, especially the R terminal is running in a tmux window and the window is in selection mode.

As far as I know, RStudio uses a self-hosted console which it has full authority on the behavior of sending code and has access to the current console input, while vscode-R has no access to state of the R terminal, at least at the moment, I double there's any possibility to make sending code more robust like RStudio if we don't use a self-hosted console like RStudio.

**Comment by assignUser** [comment]
I thought as much... Maybe something is possible via radian, I will
investigate.

Kun Ren <notifications@github.com> schrieb am Di., 18. Aug. 2020, 16:43:

**Comment by assignUser** [comment]
I think it should be possible for radian to use brackted paste mode to emulate this behavior, I have opened an issue regarding that topic: randy3k/radian#218

**Comment by renkun-ken** [comment]
Yes, I guess radian could define something in the input sequence to discard existing input.
